### PR TITLE
Split out app service plan into workspace template

### PR DESCRIPTION
**Pull request by marrobi**
# PR for issue #900

## What is being addressed

Move app service plan into base worksapce template and configure guacamole and inner eye bundles to reference this app service plan using a data element.

### CODE REVIEW

File: templates/workspace_services/guacamole/terraform/web_app.tf [hunk 1]
```diff
@@ -1,21 +1,13 @@
-resource "azurerm_app_service_plan" "guacamole" {
-  name                = "plan-${local.webapp_name}"
-  location            = data.azurerm_resource_group.ws.location
+data "azurerm_app_service_plan" "workspace" {
+  name                = "plan-${var.workspace_id }"
```
Comment (tamirkamara) [inline templates/workspace_services/guacamole/terraform/web_app.tf:2]: nit if it works

```suggestion
  name                = "plan-${var.workspace_id}"
```

File: templates/workspaces/base/terraform/appserviceplan.tf [hunk 2]
```diff
@@ -0,0 +1,14 @@
+resource "azurerm_app_service_plan" "workspace" {
+  count = var.deploy_app_service_plan ? 1 : 0
```
Comment (tamirkamara) [inline templates/workspaces/base/terraform/appserviceplan.tf:2]: Just a thought, isn't this the first time we might not have the resource we expect to?
I mean, if we try to deploy guacamole, it depends on the workspace being ready for it...

**Review by tamirkamara — APPROVED** [review]
A couple of notes/questions.

File: templates/workspaces/base/terraform/appserviceplan.tf (hunk 2 above)
Comment (marrobi) [reply templates/workspaces/base/terraform/appserviceplan.tf:2]: Yes, do ive defaulted it to true and not exposed it., so will always be there.

 Down the line after the shared service work we may look at how can update the workspace and enable the app service plan via an output of the workspace service.

**Pull request merged**